scripts/abins/input/dmol3loader.py

```diff
@@ -41,245 +55,233 @@ def read_vibrational_or_phonon_data(self):
-        :param data: Python dictionary to which k-point data should be added
+        frequencies, displacements = cls._read_mode_block(file_obj)
+        frequency_blocks = [frequencies]
+        displacement_blocks = [displacements]
+        cls._get_to_next_nonempty(file_obj)
+
+        while not (TextParser.block_end(file_obj=file_obj, msg=["STANDARD"])
+                   or TextParser.file_end(file_obj=file_obj)):
+            frequencies, displacements = cls._read_mode_block(file_obj)
+
+            frequency_blocks.append(frequencies)
+            displacement_blocks.append(displacements)
+
+            cls._get_to_next_nonempty(file_obj)
+
+        for column in range(1, 9):
+            # Check if the last displacement block has column(s) of NaN;
+            # if so number of modes was not multiple of 9 and extra columns
+            # should be removed before assembling blocks to full array
+            if np.all(np.isnan(displacement_blocks[-1][:, column])):
+                displacement_blocks[-1] = displacement_blocks[-1][:, :column]
+                break
+
+        frequencies = np.concatenate(frequency_blocks)
+        displacements = np.concatenate(displacement_blocks, axis=1)
+
+        return frequencies, displacements
+
+    @classmethod
+    def _read_mode_block(cls, file_obj) -> Tuple[np.ndarray, np.ndarray]:
+        """Get a row of frequencies and corresponding array of displacements"""
+
+        # Frequency row: "  1: freq1  2: freq2 ..."
+        TextParser.move_to(file_obj=file_obj, msg=":")
+        frequencies = np.asarray(file_obj.readline().split()[1::2],
+                                 dtype=FLOAT_TYPE)
+
+        # Displacements first row: " El x   disp_1x  disp_2x ..."
+        TextParser.move_to(file_obj=file_obj, msg=" x ")
+
+        displacement_rows = []
+        line = file_obj.readline()
+        match = _disp_row_regex.match(line.decode('utf8'))
+        while match:
+            displacement_rows.append(
+                np.asarray(match.groups(), dtype=FLOAT_TYPE))
+            match = _disp_row_regex.match(file_obj.readline()
+                                          .decode('utf8'))
+
+        if not displacement_rows:
+            raise ValueError("Displacement data not found in this block")
+        displacements = np.stack(displacement_rows)
+
+        return frequencies, displacements
+
+    @classmethod
+    def _assemble_mode_data(cls, frequencies: np.ndarray,
+                            raw_displacements: np.ndarray,
+                            masses: np.ndarray) -> dict:
+        """Process DMOL3 mode data to relevant keys and values for AbinsData
+
+        k-point information is set to single Gamma-point as this is all DMOL3
+        provides.
+
+        Displacements are also normalised here. Note that owing to a different
+        scaling/reporting convention a sqrt(mass) factor is _not_ applied as
+        with e.g. CRYSTAL17 output.
```

Comment:
```suggestion
        with other output (e.g. CRYSTAL17).
```

Reply:
Gotcha. Will word in slightly more detail than this.

Reply:
I'm just being fussy with English.